Fix `custom_dir` support and minification inconsistency

kamilkrzyskow opened the pull request:
Hello again,

my team started supporting multiple languages in the docs, and I moved the shared extra assets to our overrides directory to avoid file duplication. This created an issue with the `cache_safe` option. Seems like people don't use this setting, because otherwise this issue would be found sooner. Maybe people prefer less "invasive" ways like [this one](https://github.com/privacyguides/privacyguides.org/commit/ef9e236b2bfee731f174ef9f4eea88a876573e7d), however I'm not sure if that's all there is to it.

Additionally, I've unified the minification when using and not using the `cache_safe` option.
The order of functions makes the code still hard to read and maybe has caused the bug in the first place
My attempt at restructuring wasn't accepted, because I did too many things back then #19 and then opted into a less invasive revision, but I still think it's worth a shot.

You could add some "easy-first-commit" issues to:
- add missing tests for different cases of JavaScript files that would test them to make sure the options are correctly selected.
- simplify cache busting - however it's supposed to be added in MkDocs 1.5.0, once it releases, so I'm not sure if it's worth giving more thought into a _soooon_:tm: to be deprecated option
- restructure code to make it easier to read and flow similarly to the [event chart](https://www.mkdocs.org/dev-guide/plugins/#events)

kamilkrzyskow commented:
No feedback and no merge, and I've seen the other PR merged 🤔Am I supposed to change something here @wilhelmer?

wilhelmer commented:
To be honest, I don't know what to do with this. What's the issue with the `custom_dir`? Why would you minify anything inside the custom theme directory? Also, I'm only a super bad wannabe developer, so I can't say anything regarding missing tests or code restructuring. Maybe @byrnereese can help here.

kamilkrzyskow commented:
This PR is safe to merge, it introduces no breaking changes for the end-user.
The remarks at the end are just suggestions if you want the project to be open for "easy-first" commits. 
I can add some test cases if you don't want to wait for willing beginners, just ask 😉.

The path in my `mkdocs.yml` is the same as before:
```
  - minify:
      minify_html: !ENV [GMC_ENABLE_ON_PUBLISH, False]
      minify_css: !ENV [GMC_ENABLE_ON_PUBLISH, False]
      minify_js: !ENV [GMC_ENABLE_ON_PUBLISH, False]
      cache_safe: !ENV [GMC_ENABLE_ON_PUBLISH, False]
      htmlmin_opts:
        remove_comments: true
      css_files:
        - assets/stylesheets/extra.css
      js_files:
        - assets/javascripts/extra.js
```

The difference is that I moved all of my shared assets into the `overrides` directory:
![image](https://user-images.githubusercontent.com/34622465/224323493-90912a04-fac7-4f9c-92d2-4482c88c4e1c.png)
and then it crashed in `open()`:
https://github.com/byrnereese/mkdocs-minify-plugin/blob/cecdab80bd0741ab45a76b78937b5ab808eb5312/mkdocs_minify_plugin/plugin.py#L148-L150
because it expected that the file will be located within the `docs_dir`, but right now it's not the case, so I fixed it by finding the first valid file path in a `custom_dir` of the theme aka the `overrides` directory in my case.

The second issue was that the new `js_min` handling introduced in a previous commit was only working when the `cache_safe` setting was enabled:
https://github.com/byrnereese/mkdocs-minify-plugin/blob/e0ab4d9158d98f2ac6f0d9ec6b4623778a5d6ef4/mkdocs_minify_plugin/plugin.py#L147-L157
and you probably want to handle the `js_min` the same way even if the `cache_safe` setting is disabled 😄.
The error was hidden because the current tests didn't properly check JavaScript minification with different settings enabled.

oprypin commented on mkdocs_minify_plugin/plugin.py:
```diff
@@ -143,18 +152,24 @@ def _minify_extra_config(self, file_type: str, config: MkDocsConfig) -> None:
             if self.config["cache_safe"]:
                 docs_file_path: str = f"{config['docs_dir']}/{file_path}".replace("\\", "/")
 
+                for user_config in config.user_configs:
```
It caught me off guard that someone is actually using `user_configs` in MkDocs. I thought it's [there in MkDocs](https://github.com/search?q=repo%3Amkdocs%2Fmkdocs+user_configs&type=code) only by accident. Could you explain why it was used here, I'm curious. I thought that one could just read `config.theme` directly and get the same outcome.
